src/Symfony/Component/Security/Http/AccessToken/OidcAccessTokenHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\AccessToken;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\Exception\InvalidOidcUserException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * The token handler validates and retrieves the user identifier from an OIDC server using the token.
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+final class OidcAccessTokenHandler implements AccessTokenHandlerInterface
+{
+    private HttpClientInterface $oidcClient;
+    private ?LoggerInterface $logger;
+    private string $userProperty;
+
+    public function __construct(HttpClientInterface $oidcClient, LoggerInterface $logger = null, string $userProperty = 'email')
+    {
+        $this->oidcClient = $oidcClient;
+        $this->logger = $logger;
+        $this->userProperty = $userProperty;
+    }
+
+    public function getUserIdentifierFrom(string $accessToken): string
+    {
+        try {
+            // Call the OIDC server to retrieve the user info
+            // If the token is invalid or expired, the OIDC server will return an error
+            $userinfo = $this->oidcClient->request('GET', 'protocol/openid-connect/userinfo', [
+                'auth_bearer' => $accessToken,
+            ])->toArray();
+
+            if (empty($userinfo[$this->userProperty])) {
+                throw new InvalidOidcUserException(sprintf('"%s" property not found on OIDC server response.', $this->userProperty));
+            }
+
+            return $userinfo[$this->userProperty];
```

Comment:
The other claims may be useful (for instance to create a new user in the database). OIDC also allows requesting extra claims.

I'm not sure if there is an easy way to pass them to the calling class, but this would be useful.

Reply:
But the other claims will be useful to put in the User object. Which requires a second parse or to store this $userinfo in memory somewhere. It would be nice if passing that info could be supported natively; https://github.com/symfony/symfony/pull/48285

Reply:
I added a dependency on your PR @Jeroeny, I'll update my code according to yours once merged.
Nice work BTW!